ui-tests/src/test/java/io/syndesis/qe/steps/integrations/editor/EditorSteps.java

```diff
@@ -0,0 +1,164 @@
+    }
+
+    @When("^she adds first step between START and STEP connection$")
+    public void sheAddsFirstStep() throws Throwable {
+        flowViewComponent.clickAddStepLink(0);
+    }
+
+    @When("^she adds second step between STEP and FINISH connection$")
+    public void sheAddsSecond() throws Throwable {
+        flowViewComponent.clickAddStepLink(1);
+    }
+
+    @When("^.*adds? integration \"([^\"]*)\" on position \"([^\"]*)\"$")
+    public void addAnotherStep(String type, int stepPos) throws Throwable {
+        if (type.equalsIgnoreCase("connection")) {
+            flowViewComponent.clickAddConnectionLink(stepPos);
+        } else {
+            flowViewComponent.clickAddStepLink(stepPos);
+        }
+    }
+
+    /**
+     *
+     * unused
+     */
+    @And("^.*checks? that text \"([^\"]*)\" is \"([^\"]*)\" in hover table over \"([^\"]*)\" step$")
+    public void checkTextInHoverTable(String text, String isVisible, String stepPosition) throws InterruptedException {
+        if (isVisible.equalsIgnoreCase("visible")) {
+            Assertions.assertThat(flowViewComponent.checkTextInHoverTable(stepPosition))
+                    .isNotEmpty()
+                    .containsIgnoringCase(text);
+        } else {
+            Assertions.assertThat(flowViewComponent.checkTextInHoverTable(stepPosition))
+                    .isNotEmpty()
+                    .doesNotContain(text);
+        }
+    }
+
+    /**
+     * Every step element has class step and every option to add step/connection also has class step.
+     * So if you have 3 steps created, stepPosition is: first = 0, second = 2, third = 4 etc.
+     *
+     * @param text
+     * @param isVisible
+     * @param stepPosition
+     * @throws InterruptedException
+     */
+    @And("^.*checks? that text \"([^\"]*)\" is \"([^\"]*)\" in step warning inside of step number \"([^\"]*)\"$")
+    public void checkTextInStepWarning(String text, String isVisible, int stepPosition) throws InterruptedException {
+        if (isVisible.equalsIgnoreCase("visible")) {
+            doCheckTextInStepsWarningTable(text, stepPosition, true);
+        } else {
+            doCheckTextInStepsWarningTable(text, stepPosition, false);
+        }
+
+    }
+
+    /**
+     *
+     * ??????????????????????????????????????????????????
+     */
+
+    @And("^.*checks? that text \"([^\"]*)\" is \"([^\"]*)\" in step warning inside of steps: (.*)")
+    public void checkTextInStepsWarning(String text, String isVisible, List<String> list) throws InterruptedException {
+        for (String index : list) {
+            if (isVisible.equalsIgnoreCase("visible")) {
+                doCheckTextInStepsWarningTable(text, Integer.valueOf(index), true);
+            } else {
+                doCheckTextInStepsWarningTable(text, Integer.valueOf(index), false);
+            }
+        }
+
+    }
+
+    /**
+     *
+     * ??????????????????????????????????????????????????
```

Comment:
This method does the check from step above with different output options so instead of if-if-else-else-if-else there are 2 methods with if else

Reply:
This is a remnant of the process of moving steps to multiple step classes. We have talked about that. But thanks for the explanation.